editor/src/components/canvas/canvas-wrapper-component.tsx

```diff
@@ -1,43 +1,116 @@
+import ErrorOverlay from '../../third-party/react-error-overlay/components/ErrorOverlay'
+import Footer from '../../third-party/react-error-overlay/components/Footer'
+import Header from '../../third-party/react-error-overlay/components/Header'
 import { ReactErrorOverlay } from '../../third-party/react-error-overlay/react-error-overlay'
+import StackFrame from '../../third-party/react-error-overlay/utils/stack-frame'
+import { when } from '../../utils/react-conditionals'
+import { Button, FlexColumn, FlexRow, UtopiaTheme, useColorTheme } from '../../uuiui'
 import { setFocus } from '../common/actions'
+import { EditorAction } from '../editor/action-types'
 import {
   clearHighlightedViews,
+  clearHoveredViews,
+  clearSelection,
   openCodeEditorFile,
+  selectComponents,
+  setHighlightedViews,
+  setHoveredViews,
   setSafeMode,
   switchEditorMode,
 } from '../editor/actions/action-creators'
+import { CanvasToolbar } from '../editor/canvas-toolbar'
+import { EditorModes, isSelectModeWithArea } from '../editor/editor-modes'
+import { useDispatch } from '../editor/store/dispatch-context'
 import {
+  CanvasSizeAtom,
+  NavigatorWidthAtom,
   createCanvasModelKILLME,
   getAllCodeEditorErrors,
   getOpenUIJSFile,
   getOpenUIJSFileKey,
   parseFailureAsErrorMessages,
-  NavigatorWidthAtom,
-  CanvasSizeAtom,
 } from '../editor/store/editor-state'
 import { Substores, useEditorState } from '../editor/store/store-hook'
-import ErrorOverlay from '../../third-party/react-error-overlay/components/ErrorOverlay'
-import CloseButton from '../../third-party/react-error-overlay/components/CloseButton'
-import { fastForEach, NO_OP } from '../../core/shared/utils'
-import Footer from '../../third-party/react-error-overlay/components/Footer'
-import Header from '../../third-party/react-error-overlay/components/Header'
-import { FlexColumn, Button, UtopiaTheme, FlexRow } from '../../uuiui'
-import { useReadOnlyRuntimeErrors } from '../../core/shared/runtime-report-logs'
-import StackFrame from '../../third-party/react-error-overlay/utils/stack-frame'
-import {
-  AlwaysTrue,
-  usePubSubAtomReadOnly,
-  usePubSubAtomWriteOnly,
-} from '../../core/shared/atom-with-pub-sub'
-import { ErrorMessage } from '../../core/shared/error-messages'
 import CanvasActions from './canvas-actions'
-import { EditorModes } from '../editor/editor-modes'
 import { CanvasStrategyPicker } from './controls/select-mode/canvas-strategy-picker'
 import { StrategyIndicator } from './controls/select-mode/strategy-indicator'
-import { CanvasToolbar } from '../editor/canvas-toolbar'
-import { useDispatch } from '../editor/store/dispatch-context'
+import { windowToCanvasCoordinates } from './dom-lookup'
+import { metadataSelector, selectedViewsSelector } from '../inspector/inpector-selectors'
+import { createSelector } from 'reselect'
+
+export const CanvasWrapperTestId = 'canvas-wrapper'
+
+const possibleElementsUnderMouseSelector = createSelector(
```

Comment:
This looks like it could be potentially quite costly, have you tried it on the Before I Go project?

Reply:
totally valid point, I'd actually turn this into a function so that it would have to explicitly called before an areas select session starts

Reply:
Yeah, a bunch of these selectors have been added that only really need to be run as part of an area selection, meaning the canvas will be doing a whole bunch of extra computation for no reason 99% of the time.